frappe/core/doctype/communication/email.py

```diff
@@ -190,7 +190,7 @@ def update_parent_mins_to_first_response(doc):
 
 		# if status has a "Replied" option, then update the status for received communication
 		if ('Replied' in options) and doc.sent_or_received=="Received":
-			parent.db_set("status", "Open")
+			update_document_status_on_received_communication(parent)
```

Comment:
```suggestion
			parent.db_set("status", "Open")
			apply_assignment_rule(parent)
```
- No need for separate method
- You can import at the top as
`from <path>.assignment_rule import apply as apply_assignment_rule` to maintain readability while using the method

Reply:
Made this a separate function because this code was quite difficult to find and `update_parent_mins_to_first_response` doesn't really describe what it does. Maybe we can change the function name

Reply:
@prssanna In that case, you can rename `update_parent_mins_to_first_response` to `update_parent_document` and use docstring/comments to explain interconnected operations on parent document...

Also, I guess the method can be moved to `communication.py` where it is used.